tests/unittests/test_datasource/test_opennebula.py

```diff
@@ -886,7 +886,10 @@ def test_multiple_nics(self):
             'ETH3_SEARCH_DOMAIN': 'third.example.com third.example.org',
         }
         net = ds.OpenNebulaNetwork(
-            context, system_nics_by_mac={MAC_1: 'enp0s25', MAC_2: 'enp1s2'})
+            context,
+            mock.Mock(),
```

Comment:
Since we are modifying the call signature of read_context_disk_dir to now accept and operate on a distro instance. can we have a test that asserts distro.networking.is_physical is called for the interfaces returned by get_interfaces_by_mac and extend this existing test to assert that distro.networking.is_physical **is not** called becaues system_nics_by_mac is provided.

Reply:
> can we have a test that asserts distro.networking.is_physical is called for the interfaces returned by get_interfaces_by_mac

Yep, I can add tests for `get_physical_nics_by_mac`.

> and extend this existing test to assert that distro.networking.is_physical **is not** called becaues system_nics_by_mac is provided.

`system_nics_by_mac` is only passed by test code, so I don't think we need to test it specifically; those tests would fail if the values being passed were not used. Conversely, we already have testing that mocks `get_physical_nics_by_mac` and relies on the output it is mocked to return (e.g. `test_gen_conf_mtu`); if it weren't being called, then these tests would fail. I will add a test that `distro` is passed through appropriately, that's a gap currently.

Does that sound reasonable?

Reply:
I've addressed these two testing gaps with my latest push.